tensorflow/python/training/checkpoint_utils.py

```diff
@@ -183,16 +184,16 @@ def init_from_checkpoint(ckpt_dir_or_file, assignment_map):
     ValueError: If missing variables in current graph, or if missing
       checkpoints or tensors in checkpoints.
   """
+  init_from_checkpoint_fn = lambda strategy : _init_from_checkpoint(
```

Comment:
`strategy` is not really used in `_init_from_checkpoint` (you can see the first arg is _). So you don't actually need to pass it. 
Which is why I was proposing that you just take _ as the input for the lambda. 

```
def init_from_checkpoint(ckpt_dir_or_file, assignment_map):
  fn = lambda _: _init_from_checkpoint(ckpt_dir_or_file, assignment_map)
  if distribution_strategy_context.get_cross_replica_context():
    fn()
  else:
    distribution_strategy_context.get_replica_context().merge_call(fn)

def _init_from_checkpoint(ckpt_dir_or_file, assignment_map):
  .... # rest of the code remains the same 
```